week-5/Homework/mandatory/3-project/js/main.js

```diff
@@ -0,0 +1,141 @@
+//created new variable by selecting the different ID's using document.getElementById("blueBtn");
+let blueBtn = document.getElementById("blueBtn");
+let orangeBtn = document.getElementById("orangeBtn");
+let greenBtn = document.getElementById("greenBtn");
+let jumbotron = document.getElementsByClassName("jumbotron")[0];
+let buttons = document.querySelector(".buttons");
+let childrenButtons = buttons.children; // [<div class="child1">]
+let donateBikeBtn = childrenButtons[0];
+let volunteerBtn = childrenButtons[1];
+console.log(childrenButtons);
+blueBtn.addEventListener("click", function () {
+    jumbotron.style.background = "#588fbd";
+    donateBikeBtn.style.background = "#ffa500";
+    volunteerBtn.style.background = "black";
+    volunteerBtn.style.color = "white";
+});
+orangeBtn.addEventListener("click", function () {
+    jumbotron.style.background = "#f0ad4e";
+    donateBikeBtn.style.background = "#5751fd";
+    volunteerBtn.style.background = "#31b0d5";
+    volunteerBtn.style.color = "white";
+});
+greenBtn.addEventListener("click", function () {
+    jumbotron.style.background = "#87ca8a";
+    donateBikeBtn.style.background = "black";
+    volunteerBtn.style.background = "#8c9c08";
+});
+
+// blueBtn.addEventListener("click", blueBtnChanges) // example inc. function written with a defined name
+
+// function blueBtnChanges() {
+//     jumbotron.style.background = "#588fbd";
+//     donateBikeBtn.style.background = "#ffa500";
+//     volunteerBtn.style.background = "black";
+//     volunteerBtn.style.color = "white";
+// };
+//part 2 = form 
+let inputEmail = document.getElementById("exampleInputEmail1");
+let inputName = document.getElementById("example-text-input");
+let inputDescribeYourself = document.getElementById("exampleTextarea");
+let form = document.getElementsByTagName("form")[0]; //[0] identifies the first instance of the form 
+//when the user clicks submit - it checks email, name, describeyourself - the form should not be empty 
+form.onsubmit = function (event) {
+    let emailText = inputEmail.value;
+    let nameText = inputName.value;
+    let descriptionText = inputDescribeYourself.value;
+    let inputIsValid = false; //alert
+    // console.log("hello");
+    let newEmailResult = true;
+    let newTextResult = true;
+    let newDescriptionTextResult = true;
+    // 4th function that validates, gives an alert, and then clears the form 
+
+    //====checking email is checking and validating the conditions===
+    function checkEmailInput() {
```

Comment:
It's normally better to not `nest` functions like you have here because things can become confusing. It's normally better to have them outside each other e.g.

```
form.onsubmit = function (event) {
.....
}

function checkEmailInput() {
......
}

function nameTextInput() {
......
}
```